test/sql/types/struct/struct_concat.test

```diff
@@ -0,0 +1,70 @@
+# name: test/sql/types/struct/struct_concat.test
+# group: [struct]
+
+statement ok
+PRAGMA enable_verification
+
+query I
+SELECT struct_concat({'a': 1}, {'b': NULL}, NULL::STRUCT(k INT), struct_pack( x := 'foobar'));;
```

Comment:
Can we add a test where we don't cast the `NULL` literal (i.e. `struct_concat({'a': 1}, NULL)`)?